src/QMCWaveFunctions/Jastrow/J1Spin.h

```diff
@@ -163,13 +163,11 @@ struct J1Spin : public WaveFunctionComponent
     else
     {
       for (int i = 0; i < Nions; i++)
-        for (int j = 0; j < NumTargetGroups; j++)
-        {
-          auto igroup = Ions.getGroupID(i);
-          if (Ions.getGroupID(i) == source_type && j == target_type &&
-              J1UniqueFunctors[i * NumTargetGroups + j] == nullptr)
-            J1UniqueFunctors[igroup * Nelec + j] = std::move(afunc);
-        }
+      {
```

Comment:
I don't see why loop i is even needed.
`J1UniqueFunctors[source_type * NumTargetGroups + target_type]`

Reply:
If there were more than one combination of `i` and `j` then `afunc` would be moved multiple times. 

Looks like that is happening on [line 160](https://github.com/QMCPACK/qmcpack/pull/4612/files#diff-65688a9d941f76a01166e88ec6c78d49523743b251eeb48a87924cb6630d8e15R160) :worried:

Reply:
Very true. `target_type == -1` case seems broken. That needs to be sorted out when unifying J1 and J1Spin.